projects/counter/inc/counter.h

```diff
@@ -0,0 +1,10 @@
+#pragma once
+#include "interrupt.h"
+#include "log.h"
+#include "soft_timer.h"
+
+// Callback function for counter 1
+static void counter_a_timeout(SoftTimerId timer_id, void *context);
```

Comment:
in general we'd consider these functions private and thus would not include them in the header. private functions are prefixed `prv_`, but the specs were not 100% clear on using only one file so this is ok. Functions in the header should not be declared static.